### ISSUE TITLE
Suggestion: auto target priorities

### ISSUE THREAD

**Issue by cgf123** [OpenRA/OpenRA#3581]
would be nice if units could auto prioritise 

the prioritisation doesnt have to be too specific, but can be class-based

eg an apc/ranger would prioritise inf>buildings>vehicles, a tank would prioritise vehicles>structures>inf, an artillery/v2 would prioritise buildings>inf>vehicles

some notes:
- by default, units should attack the 1st target to come within their range
- priorities should kick in X sec only after a target has been chosen for attack, where X is the auto-target scan interval
- target priorities and scanning should only work when using attack-move or guarding (ie, when units pick targets by themselves); when the player issues a particular attack order on a target, the unit should continue shooting it until it dies or is out of range.
##

**Comment by Mailaender** [comment]
https://github.com/OpenRA/OpenRA/pull/7094 adds a priority system that targets damaged actors first.

**Comment by Mailaender** [comment]
http://cnc.wikia.com/wiki/Threat_Rating_Node_upgrade was cut from the original Tiberian Sun release.

**Comment by af** [comment]
This issue needs some +1 love and additional discussion. As mentioned, adding priority targeting would make the AI much more of an interesting challenge. Also attack moving _is a form of micro_ so it doesn't remove micro from the game at all– it just de-emphasizes the fiddliest type of micro. Besides, a veteran player will still get the best results by targeting enemies manually, if he/she is capable of it.

#2616 is a good example of where attack move is currently useless, but there are plenty of others:

* rocketmen targeting infantry instead of vehicles/air/buildings
* hinds/yaks targeting tanks instead of infantry
* tanks targeting infantry instead of vehicles

etc

**Comment by Phrohdoh** [comment]
I suggest simply auto-targeting whatever I could do the most damage to with my Versus values.

**Comment by af** [comment]
@Phrohdoh that would work
